src/compat/db_query.c

```diff
@@ -2455,8 +2455,8 @@ int
 db_query_seek_tuple (DB_QUERY_RESULT * result, int offset, int seek_mode)
 {
   int scan;
-  int rel1, rel2, rel3, rel_n;
-  int curr_tplno, tpl_cnt;
+  INT64 rel1, rel2, rel3, rel_n;
+  INT64 curr_tplno, tpl_cnt;
```

Comment:
curr_tplno gets the value from result->res.s.cursor_id.tuple_no.
but the tuple_no is int-type not INT64.
if the tuple_no can be 64 bit value, the type of tuple_no should be changed to INT64 first.